app/models/post.rb

```diff
@@ -1,3 +1,16 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+  attribute :text, :string
```

Comment:
- [ ] For all your update methods in `post`, `comment` and `like` Models. I recommend you to use `Rails Active Callbacks`- `after_save` to call those update methods every time a new post, comment and like is saved to database.

 class Post < ApplicationRecord
   ` after_save : update_post_counter`
....
    private
     def update_post_counter
      ...
    end
end

Reply:
Thank you very much sir for the review and suggested changes, cheers 🥂